src/main/java/com/emc/pravega/perf/ReaderWorker.java

```diff
@@ -0,0 +1,98 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.emc.pravega.perf;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+/**
+ *  abstract class for Readers.
+ */
+public abstract class ReaderWorker extends Worker implements Callable<Void> {
+
+    ReaderWorker(int readerId, int secondsToRun, Instant start,
+                 PerfStats stats, String readergrp, long totalEvents,
+                 int timeout) {
+        super(readerId, 0, secondsToRun,
+                false, 0, start,
+                stats, readergrp, totalEvents, timeout);
+    }
+
+    /**
+     * read the data.
+     * @throws Exception if consumer failed to read required data.
+     */
+    public abstract String readData() throws Exception;
+
+    /**
+     * close the consumer/reader.
+     *
+     */
+    public abstract void close();
+
+    /**
+     * Increment the Event counter and return the updated value
+     *  @return incremented value.
+     *
+     */
+    public abstract long eventCountIncrementAndGet();
+
+    /**
+     *  get the current event counter value
+     *  @return current value.
+     *
+     */
+    public abstract long eventCountGet();
+
+    /**
+     * read all the data from reader/consumer.
+     * @param drainStats object which collects the details about the data read till there is no data to read from stream
+     * @throws Exception if consumer failed to read required data.
+     */
+    public void cleanupEvents(PerfStats drainStats) throws Exception {
+        String ret = null;
+        do {
+            final Instant startTime = Instant.now();
+            ret = readData();
+            if (ret != null) {
+                drainStats.recordTime(null, startTime, ret.length());
+            }
+        } while (ret != null);
+        drainStats.printTotal(Instant.now());
+    }
+
+    @Override
+    public Void call() throws ExecutionException, Exception {
+        String ret = null;
+        try {
+            do {
+                ret = readData();
+                if (ret != null) {
+                    stats.recordTime(null, Instant.ofEpochMilli(Long.parseLong(ret.split(",")[0])), ret.length());
+                }
+            } while ((Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun) &&
+                    (eventCountIncrementAndGet() < totalEvents));
```

Comment:
eventCountIncrementAndGet() is an instance method that is defined above but it is being compared against a totalEvents count. 
If this is meant to be an instance variable, then it can simply be made into a local variable because it is not used in the child class.
If it is meant to be a global, then why is the child forced to implement it? there is only one possible implementation.

Reply:
removed